mcux/CMakeLists.txt

```diff
@@ -46,6 +52,27 @@ if (${MCUX_DEVICE} MATCHES "LPC|MIMXRT6")
   zephyr_library_sources(devices/${MCUX_DEVICE}/fsl_reset.c)
 endif()
 
+if (${MCUX_DEVICE} MATCHES "MIMXRT117")
+  zephyr_library_sources(devices/${MCUX_DEVICE}/fsl_romapi.c)
```

Comment:
Feels like some of these should be enabled based on driver support (for example fsl_romapi.c seems related to flash nor driver).

Reply:
ROMAPI is not a common driver file. The API functions of different soc are different, but in a specific soc we call the fsl_romapi file that provide api function.

Reply:
Do we need any special linker stuff for romapi?

Reply:
ROMAPI don't need special link stuff. It is only some API interface(address) available to the user. These functions have been specified in ROM code's address.